hash_reader_test.go

```diff
@@ -16,342 +16,211 @@
 package ionhash
 
 import (
-	"fmt"
 	"io/ioutil"
-	"reflect"
 	"testing"
 
 	"github.com/amzn/ion-go/ion"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestEmptyString(t *testing.T) {
 	ionHashReader, err := NewHashReader(ion.NewReaderStr(""), newIdentityHasherProvider())
-	if err != nil {
-		t.Fatalf("expected NewHashReader() to successfully create a HashReader; %s", err.Error())
-	}
+	require.NoError(t, err, "Expected NewHashReader() to successfully create a HashReader")
 
 	for i := 0; i < 2; i++ {
-		if !ionHashReader.Next() {
-			err = ionHashReader.Err()
-			if err != nil {
-				t.Errorf("expected ionHashReader.Next() to return false without errors; %s", err.Error())
-			}
-		}
+		assert.False(t, ionHashReader.Next())
```

Comment:
I decided to make this a bit more strict that what we had before because Next() should return false since we are evaluating an empty string.
The same test in dotnet has `Assert.AreEqual(IonType.None, ihr.MoveNext());` which is equivalent to Next() returning false in Go.

Actually `assert` does not stop executing the test. Rather `require` is used for fatal errors that should stop test execution.